webpack/assets/javascripts/react_app/redux/actions/notifications/index.js

```diff
@@ -59,7 +61,12 @@ export const startNotificationsPolling = url => (dispatch, getState) => {
   dispatch(getNotifications(url));
 };
 
-export const onMarkAsRead = (group, id) => (dispatch) => {
+export const onMarkAsRead = (group, id) => (dispatch, getState) => {
+  if (getState().notifications.notifications
+    .some(notification => notification.id === id) === false ||
+    getState().notifications.notifications
```

Comment:
1. why get state twice? probably can save it to a local var.
2. i'm not sure that reducer is the right place for it? shouldn't we not allow the action to fire to begin with?

Reply:
1. Yes, we can save it to a local var
2. Any click on one of the notification elements triggers a `MarkAsRead`, which seems right because the user interacted with the notification, If we want to change this we need to change it in PF. Thoughts?